provider/boltmem/boltmem.go

```diff
@@ -257,15 +260,47 @@ func (s *Silences) Mutes(lset model.LabelSet) bool {
 }
 
 // All returns all existing silences.
-func (s *Silences) All() ([]*types.Silence, error) {
+func (s *Silences) All() (*types.SilencesQueryResponse, error) {
+	return s.Query(len(s.k), 0)
+}
+
+// Query returns n silences starting at offset o.
+func (s *Silences) Query(n, o int) (*types.SilencesQueryResponse, error) {
 	s.mtx.RLock()
 	defer s.mtx.RUnlock()
 
-	res := make([]*types.Silence, 0, len(s.cache))
-	for _, s := range s.cache {
-		res = append(res, s)
+	klen := len(s.k)
+
+	if klen < n {
+		n = klen
 	}
-	return res, nil
+
+	pageStart := n * o
```

Comment:
`// Query returns n silences starting at offset o.` is what's currently in there. Would this be a good clarification?
```
// Query returns n silences starting at page offset o.
```